specs/latest/1.0/index.html

```diff
@@ -378,6 +378,11 @@ <h3><a name="THE_DRAWING_BUFFER">The Drawing Buffer</a></h3>
         an <a href="https://html.spec.whatwg.org/multipage/webappapis.html#imagebitmap">ImageBitmap</a> <a href="#refsHTML">[HTML]</a>
         from this context's canvas.
     </p>
+    <p>
+        When presenting the drawing buffer, the pixel values in the drawing buffer shall be
+        converted from the color space of the drawing buffer to the color space of the screen, and
+        then shall be tone mapped as specified by <code>drawingBufferToneMapping</code>.
```

Comment:
> Question 1: Why not use an attribute, like drawingBufferColorSpace?
> The IDL definition does not allow attributes to be dictionaries, so that is not an option.

Question 1b: Why WebGLToneMapping has to be a dictionary instead of the single mode entry it contains? Are there common tone mapping parameters that could be passed here in the future that could not be passed as a separate function?

Reply:
There are tone mapping modes that we plan to add which would use more parameters.

Example: HDR10. We would like to add a mode that matches HDR10 rendering (The reason it's not in this specification is because it's usually framed in relation to what would be the `rec2100-pq` color space, and we want to make sure all of the details of that are [sorted out](https://github.com/whatwg/html/issues/9112) before introducing it. We also want to do it separately to make sure that all of those details are clearly exposed for scrutiny, rather than being put in a giant bundle including other features.)

In that situation, we would want to include content light level info (CLLI) and mastering display color volume (MDCV) metadata, which are standard for use with HDR10. Over in the [explainer](https://github.com/ccameron-chromium/webgpu-hdr/blob/main/EXPLAINER.md) for the WebGPU version (which has now [merged and shipped](https://github.com/gpuweb/gpuweb/issues/4239)), we have a worked example as follows (translated to this API):

```
  gl.drawingBufferStorage(gl.RGBA16F,
                          gl.drawingBufferWidth,
                          gl.drawingBufferHeight);
  gl.drawingBufferColorSpace = "rec2100-linear";
  gl.drawingBufferToneMapping({mode:"hdr10",
                               clli:{maxFALL:200, maxCLL:1200}})
```
On macOS, this would correspond to using [HDR10 CAEDRMetadata](https://developer.apple.com/documentation/quartzcore/caedrmetadata/3194382-hdr10metadatawithdisplayinfo?language=objc), and on Windows it would correspond to using [DXGI_HDR_METADATA_HDR10](https://learn.microsoft.com/en-us/windows/win32/api/dxgi1_5/ns-dxgi1_5-dxgi_hdr_metadata_hdr10).

If such a canvas were to be then encoded to an HDR image or video stream, then the resulting image or video would be required to be include that metadata.